src/kani-compiler/src/codegen_cprover_gotoc/codegen/function.rs

```diff
@@ -263,20 +301,93 @@ impl<'tcx> GotocCtx<'tcx> {
             mangled_name,
             original_file: loc.filename().unwrap(),
             original_line: loc.line().unwrap().to_string(),
+            unwind_value: None,
         };
 
-        self.proof_harnesses.push(harness);
+        harness
+    }
+
+    /// Unwind strings of the format 'unwind(x)' and the mangled names are to be parsed for the value 'x'
+    fn handle_kanitool_unwind(&mut self, attr: &Attribute, harness: &mut HarnessMetadata) {
+        // Check if some unwind value doesnt already exist
+        if harness.unwind_value.is_none() {
```

Comment:
Same suggestion as previous comment on handling errors, then the expected case.

Reply:
Going to wrap the behavior with a match statement.